RELEASE/scripts/autoscend.ash

```diff
@@ -2351,37 +2351,44 @@ int speculative_pool_skill()
 
 boolean autosellCrap()
 {
-	if((item_amount($item[dense meat stack]) > 1) && (item_amount($item[dense meat stack]) <= 10))
+	if(can_interact() && my_meat() > 20000)
 	{
-		auto_autosell(1, $item[dense meat stack]);
+		return false;		//do not autosell stuff in casual or postronin unless you are very poor
 	}
-	foreach it in $items[Blue Money Bag, Red Money Bag, White Money Bag]
+	
+	foreach it in $items[dense meat stack, meat stack, Blue Money Bag, Red Money Bag, White Money Bag]
 	{
 		if(item_amount(it) > 0)
 		{
-			auto_autosell(item_amount(it), it);
+			auto_autosell(item_amount(it), it);		//autosell all of this item
 		}
 	}
-	foreach it in $items[Ancient Vinyl Coin Purse, Bag Of Park Garbage, Black Pension Check, CSA Discount Card, Fat Wallet, Gathered Meat-Clip, Old Leather Wallet, Penultimate Fantasy Chest, Pixellated Moneybag, Old Coin Purse, Shiny Stones, Warm Subject Gift Certificate]
+	foreach it in $items[Ancient Vinyl Coin Purse, Black Pension Check, CSA Discount Card, Fat Wallet, Gathered Meat-Clip, Old Leather Wallet, Penultimate Fantasy Chest, Pixellated Moneybag, Old Coin Purse, Shiny Stones, Warm Subject Gift Certificate]
 	{
-		if((item_amount(it) > 0) && glover_usable(it) && is_unrestricted(it))
+		if(item_amount(it) > 0 && auto_is_valid(it))
 		{
-			use(1, it);
+			use(item_amount(it), it);
 		}
 	}
-
-	if(!in_hardcore() && !isGuildClass())
+	foreach it in $items[Bag Of Park Garbage]		//keeping 1 garbage in stock to avoid possible harmful loop with dinseylandfill_garbageMoney()
 	{
-		return false;
+		if(item_amount(it) > 1)		//for these items we want to keep 1 in stock. sell the rest
```

Comment:
thanks for adding this, I know situation is somewhat unreasonable that we would be barf farming, getting park garbage, using them, and staying under 20k meat, but its also possible for that to happen or for us to introduce ways for that to happen in future code so glad to close that loop